var/spack/repos/builtin/packages/dftbplus/package.py

```diff
@@ -62,16 +76,38 @@ class Dftbplus(MakefilePackage):
         default=False,
         description="Use DftD3 dispersion library " "(if you need this dispersion model)",
     )
+    variant(
+        "api",
+        default=True,
+        description="Build the API library " "(if you need to link to DFTB+ from other codes)",
+        when="build_system=cmake",
+    )
+    variant(
+        "openmp",
+        default=True,
+        description="Build with OpenMP support " "(if you need to use OpenMP parallelization)",
+    )
+    variant(
+        "sharedlibs",
+        default=False,
+        description="Build as shared library",
+    )
 
     depends_on("lapack")
     depends_on("blas")
     depends_on("scalapack", when="+mpi")
     depends_on("mpi", when="+mpi")
     depends_on("elsi", when="+elsi")
     depends_on("magma", when="+gpu")
-    depends_on("arpack-ng", when="+arpack")
+    depends_on("arpack-ng", when="+arpack~mpi")
     depends_on("dftd3-lib@0.9.2", when="+dftd3")
+    depends_on("m4", type="build", when="@:19.1")
+    depends_on("python@3.2:", type=("build", "run"))
+    depends_on("py-numpy", type=("build", "run"))  # for tests
+    depends_on("cmake", type="build", when="@20.1:")
 
+
+class MakefileBuilder(spack.build_systems.makefile.MakefileBuilder):
```

Comment:
Are these additional classes the standard way to define the build steps if a package uses different build systems depending on its version?

Reply:
Yes, see for eg https://github.com/spack/spack/blob/c4a53cf3765258e140378326b619d5c908ee614b/var/spack/repos/builtin/packages/libtree/package.py#L10